### ISSUE TITLE
Error in rseqc_gtf2bed

### ISSUE THREAD

**Issue by alefrol638** [cnio-bu/bollito#8]
# Problem description

I am trying to run bollito on fastq files from a 10x run with v3 chemistry. However, I receive an error and the pipeline does not seem to continue.

# Expected Output
Full output from fastq files

# Error 

Error in rule rseqc_gtf2bed:
    jobid: 128
    output: /out/qc/rseqc/annotation.bed, /out/qc/rseqc/annotation.db
    log: /log/rseqc/rseqc_gtf2bed.log (check log file(s) for error message)
    conda-env: /apps/bollito/.snakemake/conda/cf27484f

RuleException:
CalledProcessError in line 80 of /apps/bollito/rules/qc.smk:
Command 'source /.conda/envs/snakemake_5.24.0/bin/activate '/apps/bollito/.snakemake/conda/cf27484f'; set -euo pipefail;  python/apps/bollito/.snakemake/scripts/tmpqegbd6vw.gtf2bed.py' returned non-zero exit status 1.
  File "/.conda/envs/snakemake_5.24.0/lib/python3.8/site-packages/snakemake/executors/__init__.py", line 2114, in run_wrapper
  File "/apps/bollito/rules/qc.smk", line 80, in __rule_rseqc_gtf2bed
  File "/.conda/envs/snakemake_5.24.0/lib/python3.8/site-packages/snakemake/executors/__init__.py", line 529, in _callback
  File "/.conda/envs/snakemake_5.24.0/lib/python3.8/concurrent/futures/thread.py", line 57, in run
  File "/.conda/envs/snakemake_5.24.0/lib/python3.8/site-packages/snakemake/executors/__init__.py", line 515, in cached_or_run
  File "/.conda/envs/snakemake_5.24.0/lib/python3.8/site-packages/snakemake/executors/__init__.py", line 2199, in run_wrapper

# System
CentOS Linux 8

Thank you!

[conda_environment.txt](https://github.com/cnio-bu/bollito/files/9699330/conda_environment.txt)

**Comment by tdido** [comment]
Hi @alefrol638.

Could you attach the log file for the step, which should be in `/log/rseqc/rseqc_gtf2bed.log` according to your error message? That may give us more information about the problem.

**Comment by alefrol638** [comment]
Thank you for the quick response!  Attached you may find the log for rseqc. 

Actually, if I remove the rseqc step from the pipeline I have an additional problem: In my library both reads have the same length of 150 bp (https://www.biostars.org/p/9529864/#9529942). Therefore, I receive an error from StarSolo, that read 1 has the wrong length (see attached log). I think this is an easy one, you could maybe just include an optional trimming step in your pipeline before the alignment. Alternatively, I think STAR v.2.7.8 supports trimming (see under "Barcode and cDNA on the same mate" in https://github.com/alexdobin/STAR/blob/master/docs/STARsolo.md#barcode-and-cdna-on-the-same-mate), so maybe setting a newer version for STAR would help. I couldn't test it, because I was not able to find the yaml, in which the STAR version is set.

[rseqc_gtf2bed.log](https://github.com/cnio-bu/bollito/files/9707426/rseqc_gtf2bed.log)
[18_4_BDJ_R1_001.fastq.gz.log](https://github.com/cnio-bu/bollito/files/9707627/18_4_BDJ_R1_001.fastq.gz.log)

**Comment by tdido** [comment]
Right, thanks for the log.

Would you mind moving the second question to a new issue? That will helps us organise the work better in order to solve both problems. Thank you in advance.

Regarding the rseqc error, it looks like the environment is being created with Python 3.10, and there's an undeclared dependency to 3.9 in the package. I've now modified the conda environment to fix this version, please test branch [8_reseqc_error](/cnio-bu/bollito/tree/8_rseqc_error) and see if the error goes away.

**Comment by alefrol638** [comment]
I have tried the pipeline with the 8_reseqc_error branch, but the error persists.

**Comment by tdido** [comment]
I haven't been able to replicate your error. Any chance you could share a minimal dataset that triggers it so we can test?

**Comment by alefrol638** [comment]
The dataset in question is publicly available under: https://zenodo.org/record/4264462. I have tested the pipeline also on our own data (it was SeqWell, though) with the same result.

**Comment by alefrol638** [comment]
I just found out, that the link I send above is TCR/BCR data only. The GEX data is not available, yet. I will keep try to prepare a small reproducible example from the SeqWell data or inform you as soon as the GEX fastqs for the publicly available dataset are available. Thank you!

**Comment by ralvarez-hub** [comment]
I've also had some error in rule rseqc_gtf2bed. I'm not sure if it's the same problem, but just in case.
Here is the log:
Traceback (most recent call last):
  File "/home/bollito/.snakemake/scripts/tmpf143srhw.gtf2bed.py", line 9, in <module>
    import gffutils
ModuleNotFoundError: No module named 'gffutils'

I tried updating gffutils version (from 0.9 to 0.11.1) in envs/gffutils.yaml, and apparently it was solved.
